include/seqan/parallel/parallel_tags.h

```diff
@@ -62,15 +63,149 @@ namespace seqan {
+ * @headerfile <seqan/parallel.h>
+ *
+ * @signature using Sequential = ExecutionPolicy<Serial, Serial>;
+ */
+using Sequential = ExecutionPolicy<>;
+
 // ============================================================================
 // Metafunctions
 // ============================================================================
 
+// ----------------------------------------------------------------------------
+// Metafunction IsVectorized
+// ----------------------------------------------------------------------------
+
+/*!
+ * @mfn ExecutionPolicy#IsVectorized
+ * @brief Evaluates to @link LogicalValuesTags#True @endlink if the execution policy is
+ *        specialized with @link ParallelismTags#Vectorial @endlink.
+ * @headerfile <seqan/parallel.h>
+ *
+ * @signature IsVectorized<TExecPolicy>::VALUE
+ * @tparam TTExecPolicy The @link ExecutionPolicy @endlink to check.
+ * @return bool <tt>true</tt> if @link ExecutionPolicy @endlink is @link ParallelismTags#Vectorial @endlink,
+ *              otherwise <tt>false</tt>.
+ */
+template <typename T>
+struct IsVectorized : False
+{};
+
+template <>
+struct IsVectorized<Vectorial> : True
+{};
+
+template <typename TPar, typename TVec>
+struct IsVectorized<ExecutionPolicy<TPar, TVec> > :
+    public IsVectorized<TVec>
+{};
+
+// ----------------------------------------------------------------------------
+// Metafunction IsParallel
+// ----------------------------------------------------------------------------
+
+/*!
+ * @mfn ExecutionPolicy#IsParallel
+ * @brief Evaluates to @link LogicalValuesTags#True @endlink if the execution policy is
+ *        specialized with @link ParallelismTags#Parallel @endlink.
+ * @headerfile <seqan/parallel.h>
+ *
+ * @signature IsParallel<TExecPolicy>::VALUE
+ * @tparam TTExecPolicy The @link ExecutionPolicy @endlink to check.
+ * @return bool <tt>true</tt> if @link ExecutionPolicy @endlink is @link ParallelismTags#Parallel @endlink,
+ *              otherwise <tt>false</tt>.
+ */
+template <typename T>
+struct IsParallel : False
+{};
+
+template <>
+struct IsParallel<Parallel> : True
+{};
+
+template <typename TPar, typename TVec>
+struct IsParallel<ExecutionPolicy<TPar, TVec> > :
+    public IsParallel<TPar>
+{};
+
+// ----------------------------------------------------------------------------
+// Metafunction IsExecutionPolicy
+// ----------------------------------------------------------------------------
+
+/*!
+ * @mfn ExecutionPolicy#IsExecutionPolicy
+ * @brief Checks if a given type is an @link ExecutionPolicy @endlink.
+ * @headerfile <seqan/parallel.h>
+ *
+ * @signature IsExecutionPolicy<T>::VALUE
+ * @tparam T The type to check.
+ * @return bool <tt>true</tt> if the <tt>T</tt> is an @link ExecutionPolicy @endlink type,
+ *              otherwise <tt>false</tt>.
+ */
+template <typename T>
+struct IsExecutionPolicy : False
+{};
+
+template <typename TPar, typename TVec>
+struct IsExecutionPolicy<ExecutionPolicy<TPar, TVec> > : True
+{};
+
```

Comment:
this looks like a lot of boilerplate....

Reply:
Well, STL is providing this metafunction as well. So we could drop this here and use this meta function of the STL as a customisation point and overload it for our types or just keep it as no harm will be added.

Reply:
But only for SeqAn 3, as c++17 is required.